src/chapters/sessions/managing-cookies.rst

```diff
@@ -0,0 +1,183 @@
+Flask Cookies
+=============
+
+When a server executes code to create a cookie, part of the command sets a time
+limit for how long to save the data. If no time is given in the code, then the
+cookie gets erased when we quit the browser. Putting our device to sleep or
+closing the tab does NOT clear cookies!
+
+Recall that Flask runs a server on our machine when we launch an application.
+This means we can include some Python code to set and access our own cookies.
+
+Let's see how to do this.
+
+Cookie Video
+------------
+
+Watch this short clip to learn how to set cookies with Flask. You can find the
+starter code in this `GitHub repository <https://github.com/LaunchCodeEducation/LCHS_cookie_intro.git>`__.
+By switching between the ``main`` to ``final-code`` branch, you can view both
+the starting and ending points of the demonstration.
+
+.. raw:: html
+
+   <section class="vid_box">
+      <iframe class="vid" src="https://www.youtube.com/embed/V-jAPj6fETg" frameborder="1" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+   </section>
+
+.. index:: ! make_response, ! set_cookie method
+
+**Video Highlights**:
+
+#. To use Flask to set cookies, stat by importing ``make_response`` from the
+   ``flask`` library.
+#. The ``make_response()`` function builds the HTTP response sent by the server
+   to the browser.
+#. In the previous Flask chapters, we sent HTTP responses with
+   ``render_template()``. ``make_response()`` lets us do the same thing, but
+   it also allows us to set cookies.
+#. The argument for ``make_response()`` can be a simple string like
+   ``"Hello, World!"``, or it can be the result returned by the
+   ``render_template()`` function.
+
+   .. sourcecode:: Python
+
+      response_name = make_response("Hello, World!")
+
+      # OR
+
+      response_name = make_response(render_template('template_name.html'))
+
+#. To set cookies, we use the (wait for it...) ``.set_cookie()`` method!
+   The general syntax for the method is:
+
+   .. sourcecode:: Python
+
+      response_name.set_cookie(cookie_name, cookie_value)
+   
+   ``response_name`` is the variable that holds the result from
+   ``make_response()``. ``cookie_name`` and ``cookie_value`` must be strings.
+
+#. The ``set_cookie()`` method can take other arguments as well. However, these
+   go beyond the scope of this course. If curious, you can find a more complete
+   description for ``set_cookie()`` `here <https://pythonise.com/series/learning-flask/flask-cookies#setting-cookies>`__.
+
+Viewing Cookies
+---------------
+
+As shown in the video, we can use our browser tools to view the cookies saved
+for a particular webpage. The specific commands will vary with each browser,
+but right-clicking on the page is a good thing to try first. Select
+*Inspect Element* from the options that pop up, and then open the *Storage*
+tab.
+
```

Comment:
I think it would be good to add a  note to send them to resources for the different browsers:

chrome - https://developer.chrome.com/docs/devtools/
 
and safari you have to download and install - https://developer.apple.com/safari/tools/

I looked and could not find a reference to link to earlier in the book

Reply:
With Chrome, right-clicking still works, but the desired tab is called "Application" instead of "Storage".  I added a new sentence to point this out.

For Safari, I added an admonition box to describe how to turn on the Dev Tools rather than having the students wade through the documentation.

Let me know what you think.

Reply:
So when I tested this Application isn't there, I had to click on >>

![image](https://user-images.githubusercontent.com/39868164/125658340-783f6c57-fc35-427c-90fd-5f2c659240b1.png)

Reply:
So maybe add the click on >> and select Application?

Reply:
As per our conversation, I updated the figure caption to encourage independent problem solving to access the browser tools.